### PR TITLE
prototype: narrator

### PR DESCRIPTION
**Pull request by asurkov**
A simple screen reader (talks in Chrome, logs in Firefox). I see how badly I need traversal API before I can do anything useful.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
looks awesome! Looking forward to playing with this!

File: prototype/html.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/**
+ * HTML semantics description in accessibility taxonomy terms.
+ *
+ * Text selector syntax:
+ *   :role a pseudo class pointing this element has to be accessible;
```
Comment (marcoscaceres) [inline prototype/html.js:5]: this is a bit confusing.

Comment (asurkov) [inline prototype/html.js:5]: the idea of :role was you could do ":role(grid) > :role(row) > td" to match a grid cell. So I thought if :role is used with no role name on 'this' element then it should mean that this element has to have an accessible element for a match. That is used to apply universal ARIA attributes like aria-label which is applicable to any accessible DOM element. Not sure for now how to make it nicer.

Comment (marcoscaceres) [inline prototype/html.js:5]: It's ok, let's roll with this. Thanks for the clarification

**Pull request merged**